src/doom/g_game.c

```diff
@@ -2628,7 +2640,7 @@ G_InitNew
 #define DEMOMARKER		0x80
 
 // [crispy] demo progress bar and timer widget
-int defdemotics = 0, deftotaldemotics;
+int defdemotics = 0, deftotaldemotics, numplayersingame;
```

Comment:
A little nitpick, but I think we can fix this without another global variable. In Woof I just moved `defdemotics++` to `G_Ticker`:
https://github.com/fabiangreffrath/woof/blob/991ed65ae8c4824d00aea176feb4d7eddc8661e1/src/g_game.c#L2267-L2268
Not sure which approach is better.

Reply:
Yes, please. One global variable less is better here.

Reply:
I'm afraid I'm not quite get it. I've made `numplayersingame` for correct length of demo progress bar / timer, while @rfomin is pointing to `playback_tic` which isn't existing in Crispy at all.

How should I/we proceed then?

Reply:
That's what I meant:
<details>
  <summary>diff</summary>

```diff
diff --git a/src/doom/g_game.c b/src/doom/g_game.c
index 4353d26d..84152964 100644
--- a/src/doom/g_game.c
+++ b/src/doom/g_game.c
@@ -1296,6 +1296,12 @@ void G_Ticker (void)
 	    } 
 	}
     }
+
+    // [crispy] increase demo tics counter
+    if (demoplayback || demorecording)
+    {
+        defdemotics++;
+    }
     
     // check for special buttons
     for (i=0 ; i<MAXPLAYERS ; i++)
@@ -2640,7 +2646,7 @@ G_InitNew
 #define DEMOMARKER		0x80
 
 // [crispy] demo progress bar and timer widget
-int defdemotics = 0, deftotaldemotics, numplayersingame;
+int defdemotics = 0, deftotaldemotics;
 // [crispy] moved here
 static const char *defdemoname;
 
@@ -2696,11 +2702,6 @@ void G_ReadDemoTiccmd (ticcmd_t* cmd)
     }
 
     cmd->buttons = (unsigned char)*demo_p++; 
-
-    // [crispy] increase demo tics counter
-    // applies to both recording and playback,
-    // because G_WriteDemoTiccmd() calls G_ReadDemoTiccmd() once
-    defdemotics++;
 } 
 
 // Increase the size of the demo buffer to allow unlimited demos
@@ -3089,11 +3090,9 @@ void G_DoPlayDemo (void)
 
     // [crispy] demo progress bar
     {
-	int i;
+	int i, numplayersingame = 0;
 	byte *demo_ptr = demo_p;
 
-	numplayersingame = 0;
-
 	for (i = 0; i < MAXPLAYERS; i++)
 	{
 	    if (playeringame[i])
@@ -3106,7 +3105,7 @@ void G_DoPlayDemo (void)
 
 	while (*demo_ptr != DEMOMARKER && (demo_ptr - demobuffer) < lumplength)
 	{
-	    demo_ptr += (longtics ? 5 : 4);
+	    demo_ptr += numplayersingame * (longtics ? 5 : 4);
 	    deftotaldemotics++;
 	}
     }
diff --git a/src/doom/hu_stuff.c b/src/doom/hu_stuff.c
index 222c445a..d95edd53 100644
--- a/src/doom/hu_stuff.c
+++ b/src/doom/hu_stuff.c
@@ -898,7 +898,7 @@ void HU_Drawer(void)
     // [crispy] demo timer widget
     if (demoplayback && (crispy->demotimer & DEMOTIMER_PLAYBACK))
     {
-	ST_DrawDemoTimer((crispy->demotimerdir ? (deftotaldemotics - defdemotics) : defdemotics) / numplayersingame);
+	ST_DrawDemoTimer(crispy->demotimerdir ? (deftotaldemotics - defdemotics) : defdemotics);
     }
     else
     if (demorecording && (crispy->demotimer & DEMOTIMER_RECORD))
```
</details>

Reply:
Got it, thanks. now it's done.